db/version_set.cc

```diff
@@ -3916,11 +3916,22 @@ Status VersionSet::ApplyOneVersionEditToBuilder(
           edit.column_family_name_);
     }
     auto cf_options = name_to_options.find(edit.column_family_name_);
-    if (cf_options == name_to_options.end()) {
+    // implicitly add persistent_stats column family without requiring user
+    // to specify
+    bool is_persistent_stats_column_family =
+        edit.column_family_name_.compare(kPersistentStatsColumnFamilyName) == 0;
+    if (cf_options == name_to_options.end() &&
+        !is_persistent_stats_column_family) {
       column_families_not_found.insert(
           {edit.column_family_, edit.column_family_name_});
     } else {
-      cfd = CreateColumnFamily(cf_options->second, &edit);
+      // recover persistent_stats CF from a DB that already contains it
+      // TODO(Zhongyi): find out a way to pass cf options
+      if (is_persistent_stats_column_family) {
+        cfd = CreateColumnFamily(ColumnFamilyOptions(), &edit);
```

Comment:
Here is a hidden contract here. We need to guarantee that we are able to open all the future persistant stats CF, and the old one. I think we need to think through what we should do to guarantee this kind of compatibility. And we need to document it in the code. Maybe it's a good idea to reserve some versioning of the persistent stats CF, so that we reserve the possibility of doing format migration in the future.

Reply:
version key added, currently will accept anything with a version value >= 0, which we will reject user-defined column family with the same name (unless it also have a version key and value which is unlikely); in the future if we ever change persistent stats key/value encoding, we will be able to read the version number and use different parsing code  in `parseKey`